Feature/login page

jacob9301 opened the pull request:
login and signup pages. will deal with cookies in separate branch.

jacob9301 committed:
modified:   app/src/app/globals.css
	new file:   app/src/app/login/page.tsx
	new file:   app/src/app/signup/page.tsx
	new file:   app/src/components/forms/LoginForm.tsx
	new file:   app/src/components/forms/SignupForm.tsx
	modified:   app/tailwind.config.js

jacob9301 committed:
modified:   app/package-lock.json
	modified:   app/package.json
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx
	modified:   app/src/components/input/Dropdown.tsx

jacob9301 committed:
style(app): login page styling

jacob9301 committed:
modified:   app/src/app/signup/page.tsx
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx

jacob9301 committed:
modified:   app/src/components/forms/LoginForm.tsx

jacob9301 committed:
modified:   app/src/components/forms/SignupForm.tsx

jacob9301 committed:
modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx

jacob9301 committed:
modified:   api/Program.cs
	new file:   app/src/api/api.ts
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx

jacob9301 committed:
modified:   .gitignore
	new file:   app/src/api/auth/login.ts
	renamed:    app/src/api/api.ts -> app/src/api/auth/postRequest.ts
	new file:   app/src/api/auth/signup.ts
	modified:   app/src/app/login/page.tsx
	modified:   app/src/app/signup/page.tsx
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx

arbrsts commented on app/src/app/globals.css:
```diff
@@ -28,4 +28,9 @@ h1, h2 {
   @apply transition-opacity opacity-0 ease-out duration-[400ms];
+}
+
+#login-form {
```
Please don't use plain css! You can do things like `w-[500px]` for `width:500`

arbrsts replied:
Or better yet, do not declare an explicit width at all

jacob9301 replied:
dont know why thats there. must have mixed up with another project

arbrsts commented on app/tailwind.config.js:
```diff
@@ -4,6 +4,7 @@ module.exports = {
     "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
     "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
     "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
+    "./src/app/login/**/*.{js,ts,jsx,tsx,mdx}",
```
We shouldn't need this,     "./src/app/**/*.{js,ts,jsx,tsx,mdx}", already targets everything in /app folder recursively

jacob9301 replied:
removed

arbrsts commented on app/src/app/login/page.tsx:
```diff
@@ -0,0 +1,21 @@
+
+export default function Login() {
+  return (
+    <div className="lg:px-12 flex flex-col h-[95vh]">
```
`h-[95vh]` is unnecessary here

jacob9301 replied:
it  stops the footer from pushing up the page content

jacob9301 replied:
fixed

arbrsts commented on app/src/app/signup/page.tsx:
```diff
@@ -0,0 +1,21 @@
+
+export default function Signup() {
+  return (
+    <div className="lg:px-12 flex flex-col h-[95vh]">
```
`h-[95vh]` is unnecessary here

arbrsts commented on app/src/components/forms/LoginForm.tsx:
```diff
@@ -0,0 +1,73 @@
+const LoginForm = () => {
+
+    return (
+        <div className='flex flex-col justify-center flex-shrink-0' >
```
What's the reason for `flex-shrink-0`? We want things to shrink based on screen size for responsiveness

jacob9301 replied:
removed

arbrsts commented on app/src/components/forms/LoginForm.tsx:
```diff
@@ -0,0 +1,73 @@
+            >{({ errors, touched }) => 
+                (<Form className='flex flex-col justify-center'>
+                    <label  htmlFor="username" className='text-gray-600 text-base font-sans font-normal leading-normal'>Username</label>
+                    {errors.username && touched.username ? <div className='text-sm font-sans font-normal leading-normal text-red-500' >{errors.username}</div> : null}
```
This will look neater if extracted into an error JSX component, it is repeated in all of the form inputs

jacob9301 replied:
done

arbrsts commented on app/src/components/forms/LoginForm.tsx:
```diff
@@ -0,0 +1,73 @@
+                    {errors.password && touched.password ? <div className='text-sm font-sans font-normal leading-normal text-red-500' >{errors.password}</div> : null}
+                    <Field type='password' id='password' name='password' className={'rounded border mb-5 ' + (errors.password && touched.password ? 'border-red-500' : 'border-gray-600')} />
+
+                    <button type='submit' className='text-xl w-350px h-63px bg-[#00A870] py-4 px-8 text-white font-serif font-bold rounded mb-5' >Login</button>
```
Remove `w-350px`, we do not want to set explicit width, and also the correct syntax is `w-[350px]`

arbrsts commented on app/src/api/auth/login.ts:
```diff
@@ -0,0 +1,11 @@
+import { RequestBody, requestOptions, sendPostRequest } from "./postRequest";
```
/api/ route I believe is for api routes https://nextjs.org/docs/pages/building-your-application/routing/api-routes

We should put all of these helper methods in maybe a folder called `util`, and also group `login.ts` and `signup.ts` together into a single class or object, e.g.

```
const Auth = {
    login: () => {
        // login code from login.ts
    },
    signup: () => {
         // etc
    }
}
```

jacob9301 replied:
done

arbrsts commented:
Let's also make the link buttons from the main page functional to the login and sign up pages

jacob9301 committed:
modified:   app/src/components/navigation/Navbar.tsx

jacob9301 commented:
navigation button works now

arbrsts commented on app/src/api/auth/postRequest.ts:
```diff
@@ -0,0 +1,42 @@
+    headers: headers
+}
+
+export const sendPostRequest = async (url: string, requestOptions: RequestInit): Promise<object> => {
```
Nice idea to have a request handler, I think it can also be improved by allowing for more than just POST requests, e.g. GET, PUT, etc.

jacob9301 replied:
you can specify the fetch type in requestOptions

jacob9301 committed:
modified:   app/src/app/globals.css
	modified:   app/src/app/login/page.tsx
	modified:   app/src/app/signup/page.tsx
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx
	modified:   app/tailwind.config.js

jacob9301 committed:
deleted:    app/src/api/auth/login.ts
	deleted:    app/src/api/auth/signup.ts
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx
	new file:   app/src/util/Auth.ts
	renamed:    app/src/api/auth/postRequest.ts -> app/src/util/fetchRequest.ts

jacob9301 committed:
new file:   app/src/app/auth/layout.tsx
	new file:   app/src/app/auth/login/page.tsx
	new file:   app/src/app/auth/signup/page.tsx
	modified:   app/src/app/layout.tsx
	deleted:    app/src/app/login/page.tsx
	modified:   app/src/app/page.tsx
	deleted:    app/src/app/signup/page.tsx
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx
	modified:   app/src/components/navigation/Navbar.tsx

jacob9301 committed:
new file:   app/src/components/forms/FormField.tsx
	modified:   app/src/components/forms/LoginForm.tsx
	modified:   app/src/components/forms/SignupForm.tsx
